### DISCUSSION TITLE
Plain `dnf` with no command, options, or arguments

### DISCUSSION THREAD

**Discussion by hrismarin** [rpm-software-management/dnf5#1716] [General]
dnf and dnf5 operate differently when running plain `dnf` with no command, options, or arguments.

In case of dnf (dnf 4) it outputs the version info and the exit code is `0`.

In case of dnf5 the output is `Missing command. Add "--help" for more information about the arguments.` and the exit code is `2`. See https://dnf5.readthedocs.io/en/latest/dnf5.8.html#exit-codes.

In the case of dnf5 is it expected (intentional) or an oversight (bug)?  If it's the latter, where is the best place to file it?

See https://github.com/osbuild/bootc-image-builder/pull/651

**Comment by ppisar** [comment]
In my opinion, the error is a feature. Why should "dnf5" display its version? Why not e.g. to display a usage help? Or perform a system upgrade instead? Or display nothing and return exit code 0? Current behavior is consistent with tools which require more arguments. E.g. "cp" command from GNU coreutils.

**Comment by hrismarin** [comment]
Thanks for the clarification. The issue is resolved in https://github.com/osbuild/bootc-image-builder/pull/658.
